default.nix

```diff
@@ -0,0 +1,161 @@
+{
+  nixpkgsPath ? ./nixpkgs.nix,
+  system ? builtins.currentSystem,
+  overlays ? [],
+  crossSystem ? (import <nixpkgs/lib>).systems.examples.musl64,
+}:
+let
+  version = "0.1.0";
+
+  # mozilla nixpkgs rust overlay
+  nixpkgs-mozilla = builtins.fetchGit {
+    url = https://github.com/mozilla/nixpkgs-mozilla;
+    ref = "master";
+    rev = "50bae918794d3c283aeb335b209efd71e75e3954";
+  };
+  rustOverlay = import "${nixpkgs-mozilla}/rust-overlay.nix";
+
+  cargo2nix = builtins.fetchGit {
+    url = https://github.com/tenx-tech/cargo2nix;
+    ref = "master";
+    rev = "7208abba93c2e92b92f0788158a0a935c39e57b8";
+  };
+  cargo2nixOverlay = import "${cargo2nix}/overlay";
+
+  # bootstrap Nixpkgs with the overlays
+  pkgs = import nixpkgsPath {
+    inherit system crossSystem;
+    overlays = overlays ++ [ rustOverlay cargo2nixOverlay ];
+  };
+
+  inherit (pkgs) lib;
+
+  # openssl supply
+  openssl =
+    pkgs:
+      pkgs.buildPackages.symlinkJoin {
+        name = "openssl";
+        paths = with pkgs.openssl; [out dev];
+      };
+
+  # macos frameworks supply, if on darwin
+  macosFrameworks = if pkgs.stdenv.isDarwin
+    then
+      with pkgs.darwin.apple_sdk.frameworks;
+      [Security CoreServices]
+    else [];
+
+  # choice of rustc
+  rustChannel = pkgs.buildPackages.rustChannelOf {
+    channel = "1.37.0";
+  };
+
+  inherit (rustChannel) cargo;
+  rustc = rustChannel.rust.override {
+    targets = [
+      (pkgs.rustBuilder.rustLib.realHostTriple pkgs.stdenv.targetPlatform)
+      "aarch64-unknown-linux-gnu"
+    ];
+  };
+
+  # source filter
+  srcFilter = {src, name, type}:
+    (type == "directory" && name == "${toString src}/overlay" -> false) &&
```

Comment:
This is okay to remove since we don't have an `overlay` directory at the root of the repo.

Reply:
Fool! Good advice